src/pages/search/index.tsx

```diff
@@ -114,24 +133,54 @@ export default function Search() {
     });
   };
 
+  const switchShowMode = () => {
+    let nextMode;
+    // If show is undefined we know we are in list view, so nextMode should be map
+    if (!show) {
+      nextMode = ShowMode.MAP;
+    } else {
+      nextMode = show === ShowMode.LIST ? ShowMode.MAP : ShowMode.LIST;
+    }
+
+    const params = getURLSearchParamsFromAsPath(router.asPath);
+    params.set("show", nextMode);
+
+    router.replace(
+      { pathname: router.pathname, query: params.toString() },
+      undefined,
+      {
+        shallow: true,
+      }
+    );
+  };
+
+  const showMode = getShowMode(show?.toString());
+
   return (
     <Page title="Search" description="Search">
-      <SearchPageSearchForm />
-      <Koros className={styles.koros} />
-
-      <Section variant="contained">
-        <SearchList
-          ref={moreResultsAnnouncerRef}
-          loading={loading}
-          onLoadMore={onLoadMore}
-          count={count}
-          blockSize={BLOCK_SIZE}
-          hasNext={pageInfo?.hasNextPage}
-          items={searchResultItems.map((item) => (
-            <SearchResultCard key={item.id} {...item} />
-          ))}
-        />
-      </Section>
+      <SearchHeader
```

Comment:
This is rather specific, but by wrapping the form in a component, it becomes more difficult to reach. This is sort of a pet peeve for me as I've dealt with a few applications with this structure and I find it makes the code harder to grasp to someone who is unfamiliar with it. On this page the integration between the search query and search filter control is rather implicit.

I would prefer that we plug the `SearchForm` into the `SearchHeader` for instance with a prop or through some other means which allows us to reach it from within this component.

Reply:
`SearchForm` is now passed down as a prop.